### PR TITLE
Not reading hook file

### LINKED ISSUE #2628: micromamba reads `~/micromamba/etc/profile.d/micromamba.sh` from older versions
When using the latest macOS homebrew build of micromamba (https://github.com/Homebrew/homebrew-core/pull/135110), some micromamba commands don't work anymore

```
$ eval "$(micromamba shell hook)" # works
$ micromamba create -n testenv python # works
[...]
$ micromamba run -n testenv python
The following argument was not expected: --shell
Run with --help for more information.
/var/folders/rz/q1r4tv0n6ll38q18fbk419mc0000gn/T/mambafCZWLOVkCne: line 5: exec: python: not found
$ micromamba activate testenv
The following argument was not expected: --shell
Run with --help for more information.
```

### PR DESCRIPTION
**Pull request by AntoinePrv**
Fix #2628. The file is still written (being conservative) but not read (edit: done ~~expect perhaps by `csh` which as a different initalization script~~).

### CODE REVIEW

File: libmamba/src/core/activation.cpp [hunk 1]
```diff
@@ -717,28 +717,28 @@ namespace mamba
...
+        if (is_cmd(this))
         {
             get_hook_contents(shell());
             return "";
         }
 
+        std::stringstream builder;
+        if (is_powershell(this) && fs::exists(hook_source_path()))
+        {
+            builder << read_contents(hook_source_path()) << "\n";
```
Comment (JohanMabille) [inline libmamba/src/core/activation.cpp:735]: So the file is still read with powershell and we still have the upgrade issue?

File: micromamba/tests/test_shell.py [hunk 2]
```diff
@@ -86,7 +86,7 @@ def test_hook(self, shell_type):
         elif shell_type == "cmd.exe":
             assert res == ""
         elif shell_type == "tcsh":
-            assert res.count(mamba_exe) == 2
+            assert res.count(mamba_exe) == 3
```
Comment (JohanMabille) [inline micromamba/tests/test_shell.py:89]: Laybe this case could be put wiht "zsh", "bash" and "posix" above? Also, what does this return code mean?

**Review by JohanMabille — APPROVED** [review]

File: libmamba/src/core/activation.cpp (hunk 1 above)
Comment (AntoinePrv) [reply libmamba/src/core/activation.cpp:735]: No it's more complicated there. I haven't quite understand what is going on.

**Comment by jonashaag** [comment]
Hooray! I'm happy to see this go.

We are still creating those files though, right? We probably can get rid of that code as well.

**Comment by AntoinePrv** [comment]
> Hooray! I'm happy to see this go.

Still missing Windows.

> We are still creating those files though, right?

Yes.

>We probably can get rid of that code as well.

I am unsure if anyone relies on them.

**Pull request merged**